core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -177,15 +181,17 @@ public ParsedQuery parse(QueryParseContext context, BytesReference source) {
         }
     }
 
-    public ParsedQuery parse(String source) throws QueryParsingException {
+    public ParsedQuery parse(String source) throws QueryShardException, QueryShardException {
```

Comment:
I know, still the class will stay but these methods should go. I would mark every single method //norelease